### ISSUE TITLE
Add expire to a consent

### ISSUE THREAD

**Issue by hezhongfeng** [spring-projects/spring-authorization-server#1311]
when I agreed to an authorization

**Expected Behavior**

After one month, we need to agree again

**Current Behavior**

Never need to re authorize

**Context**

For example, if you agree consent, it will be valid for one month.
After one month, we need to agree again

**Comment by DevDengChao** [comment]
Good idea, and I suggest you submit this feature request to OAuth working group too: https://github.com/oauth-wg/oauth-v2-1 .

**Comment by jgrandja** [comment]
@hezhongfeng The User Consent process is not defined in the OAuth2 spec. It's up to the implementor to define the process. Ultimately, this is a "product-specific" feature and should be implemented in the product - OIDC/OAuth2 server product. This project is a framework and therefore this type of feature does not belong in the framework. If the consent process was well defined in the spec, we may consider implementing it. I'm going to close this as this feature should be implemented in the product and not the framework.
